pkgs/development/libraries/glibc/common.nix

```diff
@@ -206,6 +206,8 @@ stdenv.mkDerivation ({
 
 
   '' + lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+    unset NIX_COREFOUNDATION_RPATH
```

Comment:
This is going to cause problems in multiple places, we should not set this at all if the target platform is linux. This might also be wrong for darwin->darwin cross.

Reply:
Yeah, that's fair. It looks like it's getting set here: https://github.com/NixOS/nixpkgs/blob/dbedb52ff8f200e83dd76d47ddc50c9ff869b3e3/pkgs/stdenv/darwin/default.nix#L464-L467

But that gets deep enough into the weeds of stdenv and cross infrastructure that I'm not sure how best to fix it. If possible, it would probably be better to do the target check in nix (as opposed to in bash) so we don't cause rebuilds for non-cross packages.

Reply:
The way that's currently added isn't great and was also done before most of the cross work. Most things, including the counterpart that overrides this when [frameworks](https://github.com/NixOS/nixpkgs/blob/bb7b1b59c860a23010473b0408f56c91f7d0c810/pkgs/os-specific/darwin/apple-sdk/framework-setup-hook.sh#L38-L42) are included, use `addEnvHooks` which I think would improve the situation.

Ideally this would also just be another setup-hook, just for the pure CF. The reason I added it here was to ensure that the hooks from frameworks run last. There's probably a better way to do that.

For testing it might be possible to use an `if (stdenv.hostPlatform != stdenv.buildPlatform) then addEnvHook else export` in the preHook but that might be too late. Either way the proper thing should probably go to staging.